### PR TITLE
functional Totem tileentity & village capture command & updated faction villager

### PR DESCRIPTION
**Pull request by Cheaterpaul**
recode TotemTileEntity,
removed FactionVillager for VillageProfessions & Tradelists(maybe modify profession textures),
fixed aggressive villager (maybe other texture),
moved all villagetrades to ModVillage,
converted villager schedule is 12000 ticks (a half day) turned,
village capture command,

#516 (except village worldgen)

### CODE REVIEW

File: src/main/java/de/teamlapen/vampirism/command/test/MakeVillagerAgressiveCommand.java [hunk 1]
```diff
@@ -27,9 +27,7 @@
     private static int makeVillagerAgressive(ServerPlayerEntity asPlayer) {
         List<VillagerEntity> l = asPlayer.getEntityWorld().getEntitiesWithinAABB(VillagerEntity.class, asPlayer.getBoundingBox().grow(3, 2, 3));
         for (VillagerEntity v : l) {
-            if (v instanceof IHunter || v instanceof IVampire) continue;
-            //TotemTile.makeAggressive(v, null);//TODO 1.14 village
-
+            TotemTileEntity.makeAgressive(v);
```
Comment (maxanier) [inline src/main/java/de/teamlapen/vampirism/command/test/MakeVillagerAgressiveCommand.java:34]: What if there are hunter or vampire villager?

File: src/main/java/de/teamlapen/vampirism/tileentity/TotemTileEntity.java [hunk 2]
```diff
@@ -0,0 +1,951 @@
...
+            }
+        }
+        for (ServerPlayerEntity player : oldList) {
+            captureInfo.removePlayer(player);
+        }
+    }
+
+    static {
+        captureEntities = Maps.newHashMap();
+        captureEntities.put(VReference.HUNTER_FACTION, Lists.newArrayList(new CaptureEntityEntry(ModEntities.vampire_hunter, 10), new CaptureEntityEntry(ModEntities.advanced_hunter, 2)));
```
Comment (Cheaterpaul) [inline src/main/java/de/teamlapen/vampirism/tileentity/TotemTileEntity.java:703]: Advanced hunter & advanced vampire must implement IVillageCaptureEntity

File: src/main/java/de/teamlapen/vampirism/core/ModVillage.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    }
+
+    static void registerSchedule(IForgeRegistry<Schedule> registry) {
+        registry.register(new ScheduleBuilder(new Schedule()).add(12000, Activity.IDLE).add(10, Activity.REST).add(14000, Activity.WORK).add(21000, Activity.MEET).add(23000, Activity.IDLE).build().setRegistryName(REFERENCE.MODID, "converted_default"));
+    }
+
+    /**
+     * copy of {@link VillagerTrades.ItemsForEmeraldsTrade} with damage of itemstack
+     */
+    public static class ItemsForEmeraldsTradeWithDamage implements VillagerTrades.ITrade {
```
Comment (maxanier) [inline src/main/java/de/teamlapen/vampirism/core/ModVillage.java]: core/Mode... classes should only include registrations or object holders

File: src/main/java/de/teamlapen/vampirism/core/ModVillage.java [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+    public static final VillagerProfession hunter_expert = getNull();
+    public static final VillagerProfession vampire_expert = getNull();
+
+    public static final PointOfInterestType hunter_faction = getNull();
+    public static final PointOfInterestType vampire_faction = getNull();
+
+    public static final Schedule converted_default = getNull();
+
+    public static final VillagerTrades.ITrade[] converted_trades = new VillagerTrades.ITrade[]{new VillagerTrades.EmeraldForItemsTrade(ModItems.human_heart, 9, 2, 2), new VillagerTrades.ItemsForEmeraldsTrade(ModItems.human_heart, 3, 9, 2), new ModVillage.ItemsForEmeraldsTradeWithDamage(BloodBottleIItem.getStackWithDamage(9), 1, 3, 12, 2)};
```
Comment (maxanier) [inline src/main/java/de/teamlapen/vampirism/core/ModVillage.java]: @ObjectHolder classes should not have any public static final fields that are not object holders

**Review by maxanier — COMMENTED** [review]
Noice. A lot of work again :)
How much did you test it already?

File: src/main/java/de/teamlapen/vampirism/command/test/MakeVillagerAgressiveCommand.java (hunk 1 above)
Comment (Cheaterpaul) [reply src/main/java/de/teamlapen/vampirism/command/test/MakeVillagerAgressiveCommand.java:34]: removed all faction villager

Comment (maxanier) [reply src/main/java/de/teamlapen/vampirism/command/test/MakeVillagerAgressiveCommand.java:34]: but how about already aggressive ones. or converted villager

**Comment by Cheaterpaul** [comment]
Tested a lot and only found 2 unreproducible bugs (get extended creature from Zombie entity and  a defending ivampire entity without hitbox or model)

**Comment by maxanier** [comment]
Excellent. Very cool

**Pull request merged**